### ISSUE TITLE
Compiling on Windows

### ISSUE THREAD

**Issue by MyTDT-Mysoft** [lammertb/libhttp#67]
so i'm compiling on windows using mingw 8.1 and regular cmd.exe
compiling works, altough "clean" uses linux specific stuff that fails to cleanup properly nonetheless it compiles

however i noticed a few things when compiling the simplest make lib/slib forms:

1) it requires libpthread-1.dll just for clock_gettime. so i added a clock_gettime.c to sources on windows, so that libpthread is not included if used just for that
2) its requiring a mingw gcc dll for div??? when generating the .dll as dependency that i removed by adding -static-libgcc to compile options.. so now just a warning left..

**Comment by datrilla** [comment]
I used part of code  in window vsstudio2017 it could fit  ，I didnt USE MINGW in Win.

**Comment by bugproof** [comment]
whats the point of using this possibly with less features library instead of libcurl

**Comment by MyTDT-Mysoft** [comment]
whats the point of drinking pepsi when we already have coca-cola?

**Comment by bugproof** [comment]
@MyTDT-Mysoft that's true but also maintenance is an important factor. Last commit in this repo was 12 months ago, so if something breaks you are on your own because nobody maintains it anymore. Last commit in libcurl repo was 4 hours ago. But it's also true that there aren't many HTTP clients for C.

**Comment by lammertb** [comment]
Nice conversation here :smiley: This project is not abandoned, and the source code is maintained, although not all changes are pushed straight away to the git repository.

Mingw on Windows has not been tested by me, so it may require some adaptations before it compiles smoothly.

**Comment by lammertb** [comment]
Update on this: MinGW will be supported soon.
